RangeEnhancedIterator_Generator.py

```diff
@@ -0,0 +1,90 @@
+class IteratorEnhancedRange:
+    
+    def __init__(self, *args):
+    #Arguments are: start (default = 0), end (no default, required argument), step (default = 1)
+        self.args = args
+        for arg in args:
+            if not (type(arg) == int or type(arg) == float):
+                raise Exception("Arguments must be numeric")
+        if len(args) == 1:
+            self.start = 0
+            self.end = self.args[0]
+            self.step = 1
+        elif len(args) == 2:
+            self.start = self.args[0]
+            self.end = self.args[1]
+            self.step = 1
+        elif len (args) == 3:
+            if self.args[2] == 0:
+                raise Exception("Unable to range with step = 0")
+            self.start = self.args[0]
+            self.end = self.args[1]
+            self.step = self.args[2]
+        else:
+            raise Exception("IteratorEnhancedRange requires at least one, but no more than three arguments")
+        self.startediter = False
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        if (self.step/abs(self.step))*self.start >= (self.step/abs(self.step))*self.end:
+            raise StopIteration
+        else:
+            if self.startediter is False:
+                self.startediter = True
+                return self.start
+            else:
+                if (self.step/abs(self.step))*(self.start + self.step) >= (self.step/abs(self.step))*self.end:
```

Comment:
This seems to be pretty close to 31, could we merge them?

Reply:
It can be done together with getting rid of the flag, which I would not like to do for the reasons mentioned in my comment to the line 25. Still I made analogous edit in the Generator's class. Hope that counts.